Fix operator crash race condition for CES identity map concurrent read/write

dlapcevic opened the pull request:
Cilium operator can crash in case when the `identityToCES` map is being accessed without a lock during a concurrent write. The error message is `fatal error: concurrent map read and map write`. The fix is just to add a read lock to the one location in code where the read is not guarded by a lock.

Signed-off-by: Dorde Lapcevic <[dordel@google.com](mailto:dordel@google.com)>

christarazi commented on operator/pkg/ciliumendpointslice/manager.go:
```diff
@@ -565,9 +566,12 @@ func (c *cesManagerIdentity) InsertCEPInCache(cep *cilium_v2.CoreCiliumEndpoint,
 					continue
 				}
 				defer ces.backendMutex.RUnlock()
+				defer c.identityLock.RUnlock()
```
Can you document somewhere in this function that the ordering of the locking here must always be taken this way? This will help prevent future developers from choosing the wrong ordering and causing a deadlock.

dlapcevic replied:
Done. Added above.

dlapcevic committed:
Fix operator crash race condition for CES identity map concurrent read/write

Cilium operator can crash in case when the identityToCES map is being accessed without a lock during a concurrent write. The error message is fatal error: concurrent map read and map write. The fix is just to add a read lock to the one location in code where the read is not guarded by a lock.

Signed-off-by: Dorde Lapcevic <dordel@google.com>
